andela/ah-kgl-avengers-backend
# 165020200- provide jwt token after login

## What does this PR do?
- refactors the Sign Up endpoint
- implements the local Log In endpoint
- Implements JWT functionality using PassportJS to provide a user with a token that will authenticate subsequent requests.

## Description of Task to be completed?
- Implement Sign Up of the user using the local strategy
- Implement Log In of the user using the local strategy and provide the token.

## How should this be manually tested?
Set up the project following the README.md.
Using an API client like Postman or Insomnia:
- To test the Signup endpoint Make a POST request to `localhost:3000/api/v1/auth/signup` with a JSON body that includes  `username`,`email` and `password` fields with corresponding values: 
{ "username": "username", `email` : `your.email@address.com`,  "password": `YourPassword` }
 You should get a `201 created` response if all data are valid.

- To test the Login endpoint Make a POST request to `localhost:3000/api/v1/auth/login` with a JSON body that includes `email` and `password` fields with corresponding values: 
{ "email" : `your.email@address.com`,  "password": `YourPassword` }
 You should get a `200 OK` response if all data are valid.
## Any background context you want to provide?
You will need to set up a .env file containing the Facebook and Google OAuth Client IDs and tokens.
## What are the relevant pivotal tracker stories?
[165020200](https://www.pivotaltracker.com/n/projects/2321840)
